### PR TITLE
#172 - Company submission page

### LINKED ISSUE #169: Company owner user register page
To be able to create a company, there needs to be an account for the owner.

It would be nice if we had a choice between:
 - Google Account login
 - Standard login with email and password.
 
Maybe we need personal information like full name and maybe phone number as an additional verification data?

### LINKED ISSUE #172: Company submission page
There needs to be a way to submit companies to the site. We need a form for that.

- We need a button that will take us to there. Maybe in the navigation bar or in the company browser list.
- The same form could be used to also edit company
- A way to preview information to see how it will be displayed on a list and detail page

____

Data that should be there:
 - Company name
 - Contact infomation (email, phone)
 - Description
   - Should support basic text markdown (No images, no random blinking or scrolling decorations). Maybe a special editor with preview?
 - Logo upload as described in #67 
 - Address (Country, Voivodeship, City, Street, Zip Code)
 - MapBox Map displaying coordinates for selected address
     - It should be updated when all elements for address are filled and user edits any of them
     - A button to force generate coordinates if user feels like the map didn't update correctly
     - MapBox can return few matching locations, so a dropdown with all of them would be nice.
     - Maybe allow the pin to be moved manually?
 - Combobox with specialization selection
     - Specialization combobox should update based on Field of Study and Department comboboxes.
 - Comment for a person that will verify the company. A simple text field.
     - Company can ask there for 'company has signed papers with college' badge

### PR DESCRIPTION
**Pull request by MatArasim**
This PR implements pages, that allow users to register themselves, request adding their companies and sign in to the website. 

image

image

image

Also 404 page was added to inform user about wrong URL and redirect him to home page.

image

There is also a temporary Router in Vue that will be change to Laravel one with help of Inertia, as in #180.

Closes #169, closes #172.

### CODE REVIEW

File: frontend/assets/lang/en.json [hunk 1]
```diff
@@ -15,5 +14,48 @@
...
+    },
+    "Registration":{
+        "Header": "User Registration",
+        "FirstName": "First Name",
+        "LastName": "Last Name",
+        "Password": "Password"
+    },
+    "AddCompany": {
+        "Header": "Company add request",
+        "HeaderInfo": "After filling up the form, your request will be send to moderator, where will be verify. ",
```
Comment (EwelinaLasowy) [inline frontend/assets/lang/en.json:31]:
```suggestion
        "HeaderInfo": "After filling up the form, your request will be send to moderator, where will be verified.",
```

File: frontend/assets/lang/en.json [hunk 2]
```diff
@@ -15,5 +14,48 @@
...
+    },
+    "SignIn":{
+        "Header": "Sign in to your account",
+        "Remember": "Remember me",
+        "ForgotPassword": "Forgot your password?"
+    },
+    "Registration":{
+        "Header": "User Registration",
+        "FirstName": "First Name",
+        "LastName": "Last Name",
```
Comment (EwelinaLasowy) [inline frontend/assets/lang/en.json:26]:
```suggestion
        "LastName": "Last name",
```

File: frontend/assets/lang/en.json [hunk 3]
```diff
@@ -15,5 +14,48 @@
...
         "ComboboxSpecialization": "Choose specialization"
+    },
+    "SignIn":{
+        "Header": "Sign in to your account",
+        "Remember": "Remember me",
+        "ForgotPassword": "Forgot your password?"
+    },
+    "Registration":{
+        "Header": "User Registration",
+        "FirstName": "First Name",
```
Comment (EwelinaLasowy) [inline frontend/assets/lang/en.json:25]:
```suggestion
        "FirstName": "First name",
```

File: frontend/assets/lang/en.json [hunk 4]
```diff
@@ -15,5 +14,48 @@
...
+        "LastName": "Last Name",
+        "Password": "Password"
+    },
+    "AddCompany": {
+        "Header": "Company add request",
+        "HeaderInfo": "After filling up the form, your request will be send to moderator, where will be verify. ",
+        "CompanyName": "Comapany name",
+        "Description": "Description",
+        "DescriptionInfo": "Write about your company to help students with decision.",
```
Comment (EwelinaLasowy) [inline frontend/assets/lang/en.json:34]:
```suggestion
        "DescriptionInfo": "Write about your company to help students with the decision.",
```

File: frontend/assets/lang/en.json [hunk 5]
```diff
@@ -15,5 +14,48 @@
...
+        "StreetAddress": "Street address",
+        "City": "City",
+        "Voivodeship": "Voivodeship",
+        "PostalCode": "Postal code"
+    },
+    "ForgotPassword":{
+        "Header": "Please fill email, we will send you a link to reset your password, probably",
```
Comment (EwelinaLasowy) [inline frontend/assets/lang/en.json:41]: Probably? Why? Please change this line to something more friendly.

File: frontend/assets/lang/en.json [hunk 6]
```diff
@@ -15,5 +14,48 @@
...
+        "RegistrationButton": "Register",
+        "GoHomeButton": "Go back home"
+    },
+    "CommonLabels":{
+        "Email": "Email address",
+        "Password": "Password",
+        "Country": "Country"
+    },
+    "PageNotFound":{
+        "Header": "Page not found :C",
```
Comment (EwelinaLasowy) [inline frontend/assets/lang/en.json:58]:
```suggestion
        "Header": "Page not found",
```

File: frontend/assets/lang/en.json [hunk 7]
```diff
@@ -15,5 +14,48 @@
...
+        "Description": "Please check the URL in the address bar and try again"
```
Comment (EwelinaLasowy) [inline frontend/assets/lang/en.json:59]:
```suggestion
        "Description": "Please check the URL in the address bar and try again."
```

File: frontend/assets/lang/pl.json [hunk 8]
```diff
@@ -15,5 +15,48 @@
         "ComboboxField": "Wybierz kierunek",
         "ComboboxDepartment": "Wybierz wydział",
         "ComboboxSpecialization": "Wybierz specjalizację"
+    },
+    "SignIn":{
+        "Header": "Zaloguj się na swoje konto",
```
Comment (EwelinaLasowy) [inline frontend/assets/lang/pl.json:20]: I'm not convinced to this.

File: frontend/assets/lang/pl.json [hunk 9]
```diff
@@ -15,5 +15,48 @@
...
+        "ForgotPassword": "Nie pamiętasz hasła?"
+    },
+    "Registration":{
+        "Header": "Rejstracja",
+        "FirstName": "Imię",
+        "LastName": "Nazwisko"
+    },
+    "AddCompany": {
+        "Header": "Zgłoś swoja firmę",
+        "HeaderInfo": "Po wypełnieniu formularza, twoje zgłoszenie zostanie przekazane moderatorowi do weryfikacji. ",
```
Comment (EwelinaLasowy) [inline frontend/assets/lang/pl.json:31]:
```suggestion
        "HeaderInfo": "Po wypełnieniu formularza, Twoje zgłoszenie zostanie przekazane moderatorowi do weryfikacji.",
```

File: frontend/assets/lang/pl.json [hunk 10]
```diff
@@ -15,5 +15,48 @@
...
+        "CompanyName": "Nazwa firmy",
+        "Description": "Opis",
+        "DescriptionInfo": "Napisz coś o swojej firmie aby ułatwić wybór",
```
Comment (EwelinaLasowy) [inline frontend/assets/lang/pl.json:34]:
```suggestion
        "DescriptionInfo": "Napisz coś o swojej firmie aby ułatwić studentom wybór.",
```

File: frontend/assets/lang/pl.json [hunk 11]
```diff
@@ -15,5 +15,48 @@
...
+        "StreetAddress": "Ulica",
+        "City": "Miasto",
+        "Voivodeship": "Województwo",
+        "PostalCode": "Kod pocztowy"
+    },
+    "ForgotPassword":{
+        "Header": "Podaj adres email, na którym założyłeś konto, wyślemy link resetujacy hasło",
```
Comment (EwelinaLasowy) [inline frontend/assets/lang/pl.json:41]:
```suggestion
        "Header": "Podaj adres e-mail, na którym zostało założone konto, wyślemy link resetujący hasło",
```

But we can think of something better.

File: frontend/assets/lang/pl.json [hunk 12]
```diff
@@ -15,5 +15,48 @@
...
+        "RegistrationButton": "Zarejestruj się",
+        "GoHomeButton": "Wróć na stronę główną"
+    },
+    "CommonLabels":{
+        "Email": "Adres email",
+        "Password": "Hasło",
+        "Country": "Kraj"
+    },
+    "PageNotFound":{
+        "Header": "Nie odnaleziono strony :C",
```
Comment (EwelinaLasowy) [inline frontend/assets/lang/pl.json:58]:
```suggestion
        "Header": "Nie odnaleziono strony",
```

File: frontend/assets/lang/pl.json [hunk 13]
```diff
@@ -15,5 +15,48 @@
...
+        "Description": "Sprawdź link i spóbuj ponownie"
```
Comment (EwelinaLasowy) [inline frontend/assets/lang/pl.json:59]:
```suggestion
        "Description": "Sprawdź link i spróbuj ponownie."
```

File: frontend/components/NavigationBar.vue [hunk 14]
```diff
@@ -15,10 +15,10 @@
               <a
                 href="#"
                 class="bg-gray-900 text-white px-3 py-2 rounded-md text-sm font-medium"
-                >{{ $t("NavigationBar.Map") }}
-              </a>
+                >{{ $t("NavigationBar.Map") }}</a
+              >
               <a
-                href="#"
+                href="#/addcompany"
```
Comment (EwelinaLasowy) [inline frontend/components/NavigationBar.vue:21]: Is it by purpose?

**Review by EwelinaLasowy — CHANGES_REQUESTED** [review]

File: frontend/components/NavigationBar.vue (hunk 14 above)
Comment (MatArasim) [reply frontend/components/NavigationBar.vue:21]: If you mean changing `href="#"` to `href="#/addcompany"`, then yes. Previous `href` was temporary and direct to main page, now its direct to form. Name of routes are also temporary, they will be changed with Inertia implementaiton.

File: frontend/assets/lang/en.json (hunk 5 above)
Comment (MatArasim) [reply frontend/assets/lang/en.json]: Changed to be more user friendly

File: frontend/assets/lang/pl.json (hunk 8 above)
Comment (MatArasim) [reply frontend/assets/lang/pl.json]: Ive changed it to just "Logowanie" and "Sign in"

File: frontend/assets/lang/pl.json [hunk 15]
```diff
@@ -15,5 +15,48 @@
...
+    "Buttons":{
+        "SignInButton": "Zaloguj",
+        "ChangeButton": "Zmień",
+        "CancelButton": "Zakończ",
+        "RequestButton": "Zgłoś",
+        "RegistrationButton": "Zarejestruj się",
+        "GoHomeButton": "Wróć na stronę główną"
+    },
+    "CommonLabels":{
+        "Email": "Adres email",
```
Comment (EwelinaLasowy) [inline frontend/assets/lang/pl.json:53]:
```suggestion
        "Email": "Adres e-mail",
```

File: frontend/assets/lang/pl.json [hunk 16]
```diff
@@ -15,5 +15,48 @@
...
         "ComboboxDepartment": "Wybierz wydział",
         "ComboboxSpecialization": "Wybierz specjalizację"
+    },
+    "SignIn":{
+        "Header": "Logowanie",
+        "Remember": "Zapamiętaj mnie",
+        "ForgotPassword": "Nie pamiętasz hasła?"
+    },
+    "Registration":{
+        "Header": "Rejstracja",
```
Comment (EwelinaLasowy) [inline frontend/assets/lang/pl.json:25]:
```suggestion
        "Header": "Rejestracja",
```

File: frontend/assets/lang/pl.json [hunk 17]
```diff
@@ -15,5 +15,48 @@
...
+        "FirstName": "Imię",
+        "LastName": "Nazwisko"
+    },
+    "AddCompany": {
+        "Header": "Zgłoś swoja firmę",
+        "HeaderInfo": "Po wypełnieniu formularza, Twoje zgłoszenie zostanie przekazane moderatorowi do weryfikacji.",
+        "CompanyName": "Nazwa firmy",
+        "Description": "Opis",
+        "DescriptionInfo": "Napisz coś o swojej firmie aby ułatwić studentom wybór.",
```
Comment (EwelinaLasowy) [inline frontend/assets/lang/pl.json:34]:
```suggestion
        "DescriptionInfo": "Napisz coś o swojej firmie, aby ułatwić studentom wybór.",
```

File: frontend/components/SignIn.vue [hunk 18]
```diff
@@ -0,0 +1,84 @@
...
+            <label for="remember-me" class="ml-2 block text-sm text-gray-900">
+              {{ $t("SignIn.Remember") }}
+            </label>
+          </div>
+          <div class="text-sm">
+            <a
+              href="#/forgotpassword"
+              class="font-medium text-primary hover:text-primary"
+            >
+              {{ $t("SignIn.ForgotPassword") }}?
```
Comment (EwelinaLasowy) [inline frontend/components/SignIn.vue:61]:
```suggestion
              {{ $t("SignIn.ForgotPassword") }}
```

File: frontend/assets/lang/pl.json [hunk 19]
```diff
@@ -15,5 +15,48 @@
...
+    "CommonLabels":{
+        "Email": "Adres e-mail",
+        "Password": "Hasło",
+        "Country": "Kraj"
+    },
+    "PageNotFound":{
+        "Header": "Nie odnaleziono strony",
+        "Description": "Sprawdź link i spróbuj ponownie."
     }
+
```
Comment (EwelinaLasowy) [inline frontend/assets/lang/pl.json:61]:
```suggestion
```

**Review by EwelinaLasowy — APPROVED** [review]

**Review by kamilpiech97 — APPROVED** [review]

**Pull request merged**